pallets/cash/src/core.rs

```diff
@@ -374,15 +374,26 @@ pub fn apply_chain_event_internal<T: Config>(event: ChainLogEvent) -> Result<(),
 pub fn dispatch_notice_internal<T: Config>(
     chain_id: ChainId,
     recipient_opt: Option<ChainAccount>,
+    should_increment_era: bool,
     notice_fn: &dyn Fn(NoticeId, ChainHash) -> Result<Notice, Reason>,
 ) -> Result<(), Reason> {
     // XXX this cannot fail, should not return result
     let (latest_notice_id, parent_hash) =
         LatestNotice::get(chain_id).unwrap_or((NoticeId(0, 0), chain_id.zero_hash()));
-    let notice_id = latest_notice_id.seq();
+
+    let notice_id = if should_increment_era {
+        require!(NoticeHold::iter().count() == 0, Reason::PendingEraNotice);
+        latest_notice_id.seq_era()
+    } else {
+        latest_notice_id.seq()
+    };
 
     let notice = notice_fn(notice_id, parent_hash)?; // XXX fixme cannot fail
 
+    if should_increment_era {
```

Comment:
Is this always true? Can we have new eras without a hold? I feel like this is a separate var "should hold"?